x-pack/plugins/snapshot_restore/public/app/services/http/use_request.ts

```diff
@@ -72,13 +72,21 @@ export const useRequest = ({ path, method, body, interval }: UseRequest) => {
 
   useEffect(
     () => {
+      function cleanup() {
+        isMounted.current = false;
+      }
+
       request();
+
       if (interval) {
         const intervalRequest = setInterval(request, interval);
         return () => {
+          cleanup();
           clearInterval(intervalRequest);
         };
       }
+
+      return cleanup;
```

Comment:
@jen-huang I found a couple resources [1][2] which indicate this as a pattern for cleaning up an effect. I ran into a couple console errors by leaving the Snapshots tab before the request had resolved, and adding this fixed them.

[1] https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1 
[2] https://reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect